core/ledger/kvledger/snapshot_mgmt_test.go

```diff
@@ -110,3 +117,198 @@ func TestSnapshotRequestBookKeeperErrorPaths(t *testing.T) {
 	_, err = bookkeeper2.exist(20)
 	require.Contains(t, err.Error(), "leveldb: closed")
 }
+
+func TestSnapshotRequests(t *testing.T) {
+	conf, cleanup := testConfig(t)
+	defer cleanup()
+	provider := testutilNewProvider(conf, t, &mock.DeployedChaincodeInfoProvider{})
+	defer provider.Close()
+
+	// create a ledger with genesis block
+	ledgerID := "testsnapshotrequests"
+	bg, gb := testutil.NewBlockGenerator(t, ledgerID, false)
+	gbHash := protoutil.BlockHeaderHash(gb.Header)
+	l, err := provider.Create(gb)
+	require.NoError(t, err)
+	defer l.Close()
+	kvledger := l.(*kvLedger)
+
+	// Test 1: submit a request at height 5 that is higher than block height and verify snapshot is generated when block height matches the request
+	require.NoError(t, l.SubmitSnapshotRequest(5))
+	require.Equal(t, uint64(5), kvledger.snapshotRequestBookkeeper.smallestRequestHeight)
+
+	// commit blocks until block height is 10
+	lastBlock := testutilCommitBlocks(t, l, bg, 10, gbHash)
+	// wait until snapshot height=5 is generated
+	snapshotExists := func() bool {
+		exists, err := kvledger.snapshotExists(5)
+		require.NoError(t, err)
+		return exists
+	}
+	require.Eventually(t, snapshotExists, time.Minute, 3*time.Second)
```

Comment:
I didn't get this... why it needs to be `Eventually`? when the call on line 141 returns that would mean 10 blocks have been committed which indirectly means that the snapshot at height 5 should be guaranteed to be generated.